src/main/java/de/tum/in/www1/artemis/service/util/Tuple.java

```diff
@@ -0,0 +1,41 @@
+package de.tum.in.www1.artemis.service.util;
+
+import java.util.Objects;
+
+/**
+ * Immutable tuple object.
+ * @param <X> first param.
+ * @param <Y> second param.
+ */
+public class Tuple<X, Y> {
```

Comment:
Might be interesting to use this instead: https://www.javatuples.org/index.html

Reply:
Interesting, for now I think we are ok with the simple Tuple implementation I provided, but we should keep it in mind :+1: